salad/cli.py

```diff
@@ -8,37 +9,34 @@
 from salad.terrains.everything import *
 
 BROWSER_CHOICES = [browser.lower()
-                   for browser in DesiredCapabilities.__dict__.keys()
+                   for browser in list(DesiredCapabilities.__dict__.keys())
                    if not browser.startswith('_')]
 BROWSER_CHOICES.sort()
 DEFAULT_BROWSER = 'firefox'
+DEFAULT_PLATFORM = 'Linux'
+PLATFORM_CHOICES = ['Linux', 'OS X 10.6', 'Windows XP', 'Windows 7',
+                    'Windows 8', 'Windows 8.1']
 
-class store_driver_and_version(argparse.Action):
-    drivers = BROWSER_CHOICES
-
-    def __call__(self, parser, namespace, values, option_string=None):
-        driver_info = values.split('-')
-        if driver_info[0] not in self.drivers:
-            args = {'driver': driver_info[0],
-                    'choices': ', '.join(map(repr, self.drivers))}
-            message = 'invalid choice: %(driver)r (choose from %(choices)s)'
-            raise argparse.ArgumentError(self, message % args)
-        setattr(namespace, self.dest, driver_info[0])
-        if len(driver_info) > 1:
-            setattr(namespace, 'version', driver_info[1])
-        if len(driver_info) > 2:
-            setattr(namespace, 'platform', driver_info[2].replace('_', ' '))
 
 def main(args=sys.argv[1:]):
     parser = argparse.ArgumentParser(prog="Salad",
                                      description=("BDD browswer-automation "
                                                   "made tasty."))
 
+    parser.add_argument('-V', action='store_true', default=False,
+                        help="show program's version number and exit")
+
     parser.add_argument('--browser', default=DEFAULT_BROWSER,
-                        action=store_driver_and_version, metavar='BROWSER',
+                        metavar='BROWSER',
                         help=('Browser to use. Options: %s Default is %s.' %
                               (BROWSER_CHOICES, DEFAULT_BROWSER)))
 
+    parser.add_argument('--browserversion', help=('Browser version to use.'))
+
+    parser.add_argument('--platform', default=DEFAULT_PLATFORM,
+                        help=('Platform to use. Options: %s Default is %s.' %
+                              (PLATFORM_CHOICES, DEFAULT_PLATFORM)))
+
     parser.add_argument('--remote-url',
```

Comment:
Yup! This one should date back to when I was still involved.

On 2 February 2015 at 11:31, Steven Skoczen <notifications@github.com>
wrote:

> In salad/cli.py
> <https://github.com/salad/salad/pull/60#discussion_r23916562>:
>
> >                          help=('Browser to use. Options: %s Default is %s.' %
> >                                (BROWSER_CHOICES, DEFAULT_BROWSER)))
> >
> > +    parser.add_argument('--browserversion', help=('Browser version to use.'))
> > +
> > +    parser.add_argument('--platform', default=DEFAULT_PLATFORM,
> > +                        help=('Platform to use. Options: %s Default is %s.' %
> > +                              (PLATFORM_CHOICES, DEFAULT_PLATFORM)))
> > +
> >      parser.add_argument('--remote-url',
>
> An example would be great. Does this work with any selenum url?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/salad/salad/pull/60/files#r23916562>.
>